src/ray/raylet/scheduling/local_task_manager.cc

```diff
@@ -1035,28 +1030,5 @@ uint64_t LocalTaskManager::MaxRunningTasksPerSchedulingClass(
   return static_cast<uint64_t>(std::round(total_cpus / cpu_req));
 }
 
-std::string LocalTaskManager::GetBestSchedulableNode(const internal::Work &work,
-                                                     bool spill_waiting_task,
-                                                     bool force_spillback,
-                                                     bool *is_infeasible) {
-  // If the local node is available, we should directly return it instead of
-  // going through the full hybrid policy since we don't want spillback.
-  if ((work.grant_or_reject || work.is_selected_based_on_locality ||
-       spill_waiting_task) &&
```

Comment:
hmm i think this logic is wrong. previously we don't use `spill_waiting_task` to decide if we should prioritize local node for scheduling [ref](https://github.com/ray-project/ray/blob/4ad1fba100752ed166f2a649dbe4831fbf3ae820/src/ray/raylet/scheduling/cluster_task_manager.cc#L1193)

Reply:
Do you mean we shouldn't use `spill_waiting_task` here?

Reply:
ah... i was confused. this is added in https://github.com/ray-project/ray/commit/baa14d695aafd0abb817026ddd4b4182f68f6b61